internetarchive/openlibrary
# Cache The Best Book On aspects

Closes #4611 


This PR caches aspect data that is fetched from The Best Book On (TBBO) for a day.  TBBO aspects are used to populate the book notes modals that beta-testers can open from any book page.  Aspects should not change often, so caching them will prevent a needless amount of API calls to TBBO when the book notes feature is released.

### Technical

Initially, aspects were fetched via an AJAX call every time a book notes modal was opened.  Now aspects are fetched from a new `/aspects` endpoint on OL servers, which returns cached data.  Cached aspects are configured to expire after one day.

### Testing

1. Log in to Open Library (account must be in the beta tester group).
2. Navigate to any book page.
3. Click the "What did you think?" link located below the star ratings to open the modal.
4. Take note of the last question displayed in the modal.
5. Add a new aspect in TBBO (myself or @mekarpeles can help with this).
6. Refresh the book page in Open Library, and open the modal again.
7. Ensure that the new aspect is not displayed in the modal.
8. Delete the aspect that was added in TBBO.

### Screenshot


### Stakeholders

@mekarpeles 


### Important Note:
YAML configurations must be updated when this is deployed.
